src/main/java/cs2103/v15_1j/jimjim/searcher/DateTimeFilter.java

```diff
@@ -24,8 +29,26 @@ public LocalDateTime getEnd() {
 
     @Override
     public boolean check(TaskEvent taskEvent) {
-        // TODO Auto-generated method stub
-        return false;
+        if (taskEvent instanceof FloatingTask) {
+        	return false; // floating tasks have no datetime
+        } else if (taskEvent instanceof DeadlineTask) {
+        	DeadlineTask deadlineTask = (DeadlineTask) taskEvent;
+        	LocalDateTime deadline = deadlineTask.getDateTime();
+        	if ((start.compareTo(deadline) <= 0 ) && (end.compareTo(deadline) >= 0)) {
+        		return true;
+        	} else return false;
+        } else {
+        	Event event = (Event) taskEvent;
+        	List<EventTime> eventTimes = event.getDateTimes();
+        	for (EventTime eventTime : eventTimes) {
+        		LocalDateTime eventStart = eventTime.getStartDateTime();
+        		LocalDateTime eventEnd = eventTime.getEndDateTime();
+        		if ((start.compareTo(eventStart) <= 0) && (end.compareTo(eventEnd) >= 0)) {
```

Comment:
You are checking whether (eventStart, eventEnd) is within (start, end). I think it should be whether the 2 ranges overlap